pkg/server/controllers.go

```diff
@@ -829,6 +831,51 @@ func (s *Server) installVirtualWorkspaceURLsController(ctx context.Context, conf
 	return nil
 }
 
+func (s *Server) installKubeQuotaController(ctx context.Context, config *rest.Config, server *genericapiserver.GenericAPIServer) error {
+	controllerName := "kcp-kube-quota-controller"
+	config = rest.AddUserAgent(config, controllerName)
+
+	kubeClusterClient, err := kubernetes.NewClusterForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	// TODO(ncdc): should we make these configurable?
+	// TODO(ncdc): see if these are interpreted correctly from upstream.
+	const (
+		resyncPeriod             = 5 * time.Minute
+		replenishmentPeriod      = 12 * time.Hour
+		workersPerLogicalCluster = 5
+	)
+
+	c := kubequota.NewController(
+		s.kcpSharedInformerFactory.Tenancy().V1alpha1().ClusterWorkspaces(),
+		kubeClusterClient,
+		s.kubeSharedInformerFactory.Core().V1().ResourceQuotas(),
+		informerfactory.NewInformerFactory(s.kubeSharedInformerFactory, s.metadataSharedInformerFactory),
+		resyncPeriod,
+		replenishmentPeriod,
+		workersPerLogicalCluster,
+		s.syncedCh,
+	)
+
+	if err := server.AddPostStartHook(controllerName, func(hookContext genericapiserver.PostStartHookContext) error {
+		if err := s.waitForSync(hookContext.StopCh); err != nil {
+			klog.Errorf("failed to finish post-start-hook %s: %v", controllerName, err)
+			// nolint:nilerr
+			return nil // don't klog.Fatal. This only happens when context is cancelled.
```

Comment:
You could `errors.Is(context.Cancelled)` if you wanted to be sure

Reply:
please don't change this here. This code is everywhere. Either change all locations or non. We know there is only this error that can happen. PostStartHooks have braindead handling of errors. Returning an error is of no use, but harmful (= distracts from the actual problem during startup).